pkg/api/types.go

```diff
@@ -2353,6 +2363,20 @@ type ServiceSpec struct {
 	// cloud-provider does not support the feature."
 	// +optional
 	LoadBalancerSourceRanges []string
+
+	// externalTraffic denotes if this Service desires to route external traffic to
+	// local endpoints only. This preserves Source IP and avoids a second hop for
+	// LoadBalancer and Nodeport type services.
+	// +optional
+	ExternalTraffic ServiceExternalTrafficType
```

Comment:
I guess "" is appropriate for those cases.  Defaulting (and update) should detect LoadBalancer and NodePort and choose "Global", I think.  Thoughts?  If we do that then we really should change it back to "" if the type is updated to anything else.